### ISSUE TITLE
Difference between sklearn R^2 and duckdb R^2

### ISSUE THREAD

**Issue by cabreraalex** [duckdb/duckdb#13100]
### What happens?

When calculating R^2 using duckdb I get a (significantly) different result than when using the sklearn implementation. For the example below:

```
R^2 between the two distributions scikit: 0.7393
R^2 between the two distributions duckdb: 0.7874
```

### To Reproduce

```python
import numpy as np
from sklearn.metrics import r2_score
import duckdb

conn = duckdb.connect()

np.random.seed(42) 

mean, std = 0, 1
dist1 = np.random.normal(mean, std, 1000)

noise = np.random.normal(0, 0.5, 1000)
dist2 = dist1 + noise

r2_sklearn = r2_score(dist1, dist2)

df = pd.DataFrame({"dist1": dist1, "dist2": dist2})

r2_duckdb = conn.query("""SELECT regr_r2(dist2, dist1) as r2 FROM df""").to_df()['r2'][0]

print(f"\nR^2 between the two distributions scikit: {r2_sklearn:.4f}")
print(f"\nR^2 between the two distributions duckdb: {r2_duckdb:.4f}")
```

### OS:

darwin

### DuckDB Version:

1.0.0

### DuckDB Client:

Python

### Full Name:

Alex Cabrera

### Affiliation:

CMU

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have not tested with any build

### Did you include all relevant data sets for reproducing the issue?

Yes

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by cabreraalex** [comment]
Alright I think I found the difference. `regr_r2` is the single regression little r^2, or pearson `corr` squared. This is different than the definition from wikipedia on the https://en.wikipedia.org/wiki/Coefficient_of_determination . 

So technically correct but a bit confusing, maybe worth clarifying in the docs.

**Comment by soerenwolfers** [comment]
@cabreraalex  Wikipedia actually mentions that there is ambiguity in the definition of R2: 

```
Under more general modeling conditions, where the predicted values might be generated from a model different from linear least squares regression, an R2 value can be calculated as the square of the [correlation coefficient](https://en.wikipedia.org/wiki/Pearson_product-moment_correlation_coefficient) between the original 
y and modeled f data values. In this case, the value is not directly a measure of how good the modeled values are, but rather a measure of how good a predictor might be constructed from the modeled values (by creating a revised predictor of the form α + βƒi). According to [Everitt](https://en.wikipedia.org/wiki/Coefficient_of_determination#cite_note-10) this usage is specifically the definition of the term "coefficient of determination": the square of the correlation between two (general) variables.
``

TL;DR: You can either define R2(Y, X) using the actual residuals Y_i - X_i or using the "residuals if the best linear rescaling of X is used".

Duckdb's functions follows PostgreSQL https://www.postgresql.org/docs/9.4/functions-aggregate.html

I'd propose making the current behavior more clear in the documentation and potentially adding a new `r2_score` function that behaves like scikit.

**Comment by soerenwolfers** [comment]
I made a PR to improve the documentation: https://github.com/duckdb/duckdb-web/pull/3325

If `r2_score` does get added, a reference to it could be added there as well, and it could be described as `The raw coefficient of determination, computed using the residuals y_i - x_i without rescaling or shifting x.`

**Comment by cabreraalex** [comment]
Awesome thanks! Would be cool to have r2_score eventually
